client.go

```diff
@@ -33,24 +33,33 @@ var (
 	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
+// Mender API Client wrapper. A standard http.Client is compatible with this
+// interface and can be used without further configuration where ApiClient is
+// expected. Instead of instantiating the client by yourself, one can also use a
+// wrapper call NewApiClient() that sets up TLS handling according to passed
+// configuration.
+type ApiClient interface {
```

Comment:
I would stick to the Go naming convention here with `er` suffix. we should have `doer interface`.